tests/test_parse.py

```diff
@@ -189,20 +193,3 @@ def test_file_pickle_no_ext(tmpdir):
     p = tmpdir.join('test')
     p.write_binary(pickle.dumps(dict(a=12, b=8)))
     assert Model.parse_file(str(p), content_type='application/pickle', allow_pickle=True) == Model(a=12, b=8)
-
-
-@pytest.mark.xfail(reason='working on V2')
-def test_const_differentiates_union():
-    class SubModelA(BaseModel):
-        key: str = Field('A', const=True)
-        foo: int
-
-    class SubModelB(BaseModel):
-        key: str = Field('B', const=True)
-        foo: int
-
-    class Model(BaseModel):
-        a: Union[SubModelA, SubModelB]
-
-    m = Model.model_validate({'a': {'key': 'B', 'foo': 3}})
-    assert isinstance(m.a, SubModelB)
```

Comment:
Question 10: Since we no longer support `const` on `Field`, and we have plenty of tests of `Literal`, I don't think this test is necessary any more, but let me know if it should somehow be retained.

If we were okay updating it to use Literal, we could do that; if we wanted to keep these type hints, we'd need to use field validators. However, I think both approaches are thoroughly tested elsewhere, which is why I want to drop this test.

Reply:
That is fine with me

Reply:
remove.